Move `ImmuTable` to dedicated package

mokagio opened the pull request:
I was looking around the codebase and saw what looked like a quick win for modularization: move the `ImmuTable` implementation in a dedicated package.

This is just a little step forward towards more modularization and less time spend compiling. 

WordPressUI could have been another destination for the logic. If you feel it's better to move it there, I'll follow up with a dedicated PR. The bulk of the work was to move it out of the main target anyway

To test:

## Regression Notes
1. Potential unintended areas of impact


2. What I did to test those areas of impact (or what existing automated tests I relied on)


3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

mokagio committed:
Update `.editorconfig` to indent Swift with 4 spaces outside Xcode

mokagio committed:
Move `ImmuTableCell` to new Swift package `ImmuTable`

mokagio committed:
Move `ImmuTableRow` to `ImmuTable` package

dangermattic commented:
<!--
  0 Errors
  2 Warnings: <code>Modules/Package.swift</c...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>Modules/Package.swift</code> was changed without updating its corresponding <code>Package.resolved</code>. Please resolve the Swift packages as appropriate to your project setup (e.g. in Xcode or by running <code>swift package resolve</code>).</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Move `ImmuTable` plus a few dependencies to the `ImmuTable` package

mokagio committed:
Move the remaining bulk of `ImmuTable` in package

It was easier than identifying the access control relationships, change
them for the purpose of moving the files, then updating once the move
was completed.

mokagio committed:
Update WordPressTest imports and source to build with new ImmuTable

I run the extraction building the Jetpack target which clearly misses
some of the tests.

mokagio commented:
FYI @kean @crazytonyli I started this when first looking into this round of modularization but then paused it for the Core Data stuff. I don't consider it urgent but wanted to let you know about it in case you end up touching the `ImmuTable-` pieces.

kean commented:
Oh looks like I removed some of the code that was using it. We don't really use this type anymore. For forms, we stick with SwiftUI.

mokagio commented:
> Oh looks like I removed some of the code that was using it.
Excellent

> For forms, we stick with SwiftUI.
_This is the way_

mokagio commented:
I'll see whether to continue with this or drop once done with #24165

kean commented:
I'm going to close this PR as we've wrapped up the modularization effort and it also seems too far behind trunk now.
